rhino/src/main/java/org/mozilla/javascript/RegExpProxy.java

```diff
@@ -13,10 +13,8 @@
  */
 public interface RegExpProxy {
     // Types of regexp actions
-    public static final int RA_MATCH = 1;
     public static final int RA_REPLACE = 2;
     public static final int RA_REPLACE_ALL = 3;
-    public static final int RA_SEARCH = 4;
```

Comment:
if we really like to cleanup this i think we can drop all 4 constants and use a boolean parameter to for replace_all. This might show a lot more places where we can simplify the code.